File: 백준/Gold/13023. ABCDE/ABCDE.py
import sys 
input = sys.stdin.readline
# sys.setrecursionlimit(10**6)
n , m = map(int,input().split())
A = [
    [] for _ in range(n)
]
visited = [False]*2001
arive = False

def DFS(now, depth):
    global arive
    if depth== 5 or arive:
        arive = True
        return
        
    visited[now] = True

    for i in A[now]:
        if not visited[i]:
            # visited[i] =True
            DFS(i,depth+1)
    visited[now] = False

for i in range(m):
    s, e = map(int,input().split())
    A[s].append(e)
    A[e].append(s)

# print(A)

for i in range(n):
    DFS(i,1)  
    # visited[i] = False
    if arive:
        break
if arive:
    print(1)
else:
    print("0")
 

File: 백준/Gold/13023. ABCDE/ABCDE.swift
func Q_13023() {
    let NM = readLine()!.split { $0 == " " }.map { Int($0)! }
    let (N,M) = (NM[0], NM[1])
    
    var grid = Array(repeating: [Int](), count : N)
    var visited = Array(repeating: false, count : N)
    var _fin = false
    
    
    for _ in 0 ..< M {
        let AB = readLine()!.split { $0 == " " }.map { Int($0)! }
        let (A,B) = (AB[0], AB[1])
        grid[A].append(B)
        grid[B].append(A)
    }
    
    
    func dfs(_ x: Int, _ y : Int) {
        if y == 5 || _fin {
            _fin = true
            return
        }
        visited[x] = true
        for i in grid[x] {
            if !visited[i] {
                dfs(i, y + 1)
            }
        }
        visited[x] = false
    }
    
    for i in 0 ..< N {
        dfs(i,1)
        if _fin {
            break
        }
    }
    print(_fin ? 1 : 0)
    
}
Q_13023()